src/middlewared/middlewared/plugins/jail.py

```diff
@@ -501,6 +502,70 @@ def defaults(self, options):
             for branch in re.findall(r'refs/heads/(.*)', cp.stdout)
         }
 
+    @periodic(interval=86400)
+    @private
+    def periodic_plugin_update(self):
+        plugin_available = self.middleware.call_sync('plugin.available')
+        plugin_available.wait_sync()
+        self.middleware.call_sync('plugin.plugin_updates')
+
+    @private
+    @job(lock='plugin_updates')
+    def plugin_updates(self, job):
+        installed_plugins = self.query()
+        if not installed_plugins:
+            return
+
+        # There are 2 cases, one is where we check the version of the installed plugin with available, next is
+        # we check the plugin manifest version with the installed plugin's manifest. If any of the case is true,
+        # we raise an alert for the plugin update
+        git_repos = {}
+        for plugin in installed_plugins:
+            repo = plugin['plugin_repository']
+            if repo not in git_repos:
+                git_repos[repo] = self.middleware.call_sync('plugin.available', {'plugin_repository': repo})
+
+        for repo, job_obj in git_repos.items():
+            job_obj.wait_sync()
+            if job_obj.error:
+                self.middleware.logger.error(f'Failed to retrieve plugin versions for {repo}: {job_obj.error}')
+                git_repos[repo] = []
+            else:
+                git_repos[repo] = job_obj.result
+
+        for plugin in installed_plugins:
+            repo = plugin['plugin_repository']
+            plugin_dict = (list(filter(lambda d: d['name'] == plugin['plugin'], git_repos[repo])) or [{}])[0]
+            if not plugin_dict or any(
+                plugin_dict[k] == 'N/A' or plugin[k] == 'N/A' for k in ('version', 'revision', 'epoch')
```

Comment:
didnt we remove version from plugins json?

Reply:
`plugin_dict` here is data retrieved from `plugin.available` and `plugin` here is data retrieved from `plugin.query`. Both are guaranteed to give pkg version/revision/epoch keys.